src/contracts/middleware/StakeRegistry.sol

```diff
@@ -0,0 +1,481 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+import "@openzeppelin/contracts/utils/math/Math.sol";
+import "../interfaces/IServiceManager.sol";
+import "../interfaces/IStakeRegistry.sol";
+import "../interfaces/IRegistryCoordinator.sol";
+import "./StakeRegistryStorage.sol";
+
+/**
+ * @title A `Registry` that keeps track of stakes of operators for up to 256 quroums.
+ * Specifically, it keeps track of
+ *      1) The stake of each operator in all the quorums they are apart of for block ranges
+ *      2) The total stake of all operators in each quorum for block ranges
+ *      3) The minimum stake required to register for each quorum
+ * It allows an additional functionality (in addition to registering and deregistering) to update the stake of an operator.
+ * @author Layr Labs, Inc.
+ */
+contract StakeRegistry is StakeRegistryStorage {
+    // EVENTS
+    /// @notice emitted whenever the stake of `operator` is updated
+    event StakeUpdate(
+        bytes32 indexed operatorId,
+        uint8 quorumNumber,
+        uint96 stake
+    );
+
+    constructor(
+        IRegistryCoordinator _registryCoordinator,
+        IStrategyManager _strategyManager,
+        IServiceManager _serviceManager
+    ) StakeRegistryStorage(_registryCoordinator, _strategyManager, _serviceManager)
+    // solhint-disable-next-line no-empty-blocks
+    {
+    }
+
+    /**
+     * @notice Adds empty first entries to the dynamic arrays `_totalStakeHistory`,
+     * to record an initial condition of zero operators with zero total stake.
```

Comment:
looks like this comment is no longer accurate? (unless I'm missing something?)

Reply:
https://github.com/Layr-Labs/eigenlayer-contracts/pull/76/commits/1a690fa781e87e1c7a88da5f6b80e45f566752af